### PR TITLE
JENKINS-46579 -  Concurrency problems with spy log

### PR DESCRIPTION
**Pull request by alvarolobato**
Add some tests to check concurrency issues and a fix to make sure file is unique.

cc - @cyrille-leclerc

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
 # Pipeline Maven Plugin
-This plugin provides integration with Pipeline, configures maven environment to use within a pipeline job by calling `sh mvn` or `bat mvn`.
+This plugin provides maven integration with Pipeline through the `withMaven` step. Configures maven environment to use within a pipeline job by calling `sh mvn` or `bat mvn`.
```
Comment (alvarolobato) [inline README.md]: I messed up with my previous PR, this should be innocuous though.

File: maven-spy/src/main/java/org/jenkinsci/plugins/pipeline/maven/eventspy/reporter/FileMavenEventReporter.java [hunk 2]
```diff
@@ -74,7 +76,7 @@ public FileMavenEventReporter() throws IOException {
         }
 
         String now = new SimpleDateFormat("yyyyMMdd-HHmmss-S").format(new Date());
-        outFile = new File(reportsFolder, "maven-spy-" + now + ".log");
+        outFile = File.createTempFile("maven-spy-" + now, ".log", reportsFolder);
```
Comment (cyrille-leclerc) [inline maven-spy/src/main/java/org/jenkinsci/plugins/pipeline/maven/eventspy/reporter/FileMavenEventReporter.java]: very smart

File: maven-spy/src/main/java/org/jenkinsci/plugins/pipeline/maven/eventspy/reporter/FileMavenEventReporter.java [hunk 3]
```diff
@@ -85,6 +87,7 @@ public FileMavenEventReporter() throws IOException {
         } catch (IOException e) {
             throw new RuntimeIOException(e);
         }
+        isOpen = true;
```
Comment (cyrille-leclerc) [inline maven-spy/src/main/java/org/jenkinsci/plugins/pipeline/maven/eventspy/reporter/FileMavenEventReporter.java]: why do you need this `isOpen` flag? It's the only constructor, I don't understand how `isOpe`  could be false.

**Comment by alvarolobato** [comment]
It's for the close, it shouldn't fail if you call it twice

**Pull request merged**